Automattic/jetpack
# Business Hours block: Update PHP tests to use parent class, update fixture docs

Related to: https://github.com/Automattic/jetpack/issues/19812

Following on from https://github.com/Automattic/jetpack/pull/19743 and https://github.com/Automattic/jetpack/pull/19985, this PR refactors the PHP tests for the Business Hours block to use the new parent class responsible for dealing with server-rendered block fixtures. Now that we have the parent `Jetpack_Block_Fixture_TestCase` class in place, I've updated the block fixtures doc to reflect how fixtures tests for dynamic blocks work.

#### Changes proposed in this Pull Request:

* Refactor Business Hours block PHP tests to use parent `Jetpack_Block_Fixture_TestCase` class
* Update block fixtures doc to reflect the approach for testing server-rendered blocks

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






1. Do the updated docs read okay?

2. Ensure this doesn't introduce any regressions:

* Check out this PR
* Run the following to ensure the block extensions tests pass:

```
pnpm docker:phpunit -- --testsuite=extensions
```

* Delete the following fixtures:
  * `projects/plugins/jetpack/extensions/blocks/business-hours/test/fixtures/jetpack__business-hours.server-rendered.html`
  * `projects/plugins/jetpack/extensions/blocks/business-hours/test/fixtures/jetpack__business-hours__custom-hours.server-rendered.html`

* Re-run the above command, and the tests should fail, but report back that the fixtures have been generated
* Run `git status` to check that the generated fixtures are as they were before
* Re-run the above command to make sure that the tests pass
